src/run_program.rs

```diff
@@ -312,11 +329,17 @@ where
         &mut self,
         program: &T::Ptr,
         args: &T::Ptr,
-        max_cost: Cost,
+        mut max_cost: Cost,
     ) -> Response<T::Ptr> {
         self.val_stack = vec![self.allocator.new_pair(program.clone(), args.clone())?];
         self.op_stack = vec![Operation::Eval];
 
+        if max_cost == 0 {
+            max_cost = Cost::MAX;
```

Comment:
with this patch, `max_cost` must always be set, this is how integer wrap-arounds are prevented

Reply:
setting it to the max value effectively means infinite. I'm kind of thinking that we might want a more reasonable default than this though.

Reply:
> max_cost must always be set, this is how integer wrap-arounds are prevented
> setting it to the max value effectively means infinite

You should comment that

Reply:
yeah, I should. I should also change this code to be more true to modern programming practices:

```
let max_cost = if max_cost == 0 { Cost::MAX } else { max_cost };
```

Reply:
(so `max_cost` can be left immutable)